### PR TITLE
chore(atlas-service): convert to plugin

### PR DESCRIPTION
**Pull request by mabaasit**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass/src/app/index.tsx [hunk 1]
```diff
@@ -91,6 +94,59 @@ import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
...
+      <LoggerAndTelemetryProvider value={loggerProviderValue}>
+        {children}
+      </LoggerAndTelemetryProvider>
+    </PreferencesProvider>
+  );
+};
+
+const WithAtlasProviders: React.FC = ({ children }) => {
+  return (
+    <AtlasAuthServiceProvider value={new CompassAtlasAuthService()}>
```
Comment (gribnoysup) [inline packages/compass/src/app/index.tsx:113]: We kinda know this never re-renders after initial render here, but just to be absolutely sure our component never creates a new instance after mounting

```suggestion
  const authService = useRef(new CompassAtlasAuthService())
  return (
    <AtlasAuthServiceProvider value={authService.current}>
```

File: packages/compass/src/app/index.tsx [hunk 2]
```diff
@@ -91,6 +94,59 @@ import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
...
+  const recentQueryStorageProviderValue: RecentQueryStorageAccess = {
+    getStorage(options) {
+      return new CompassRecentQueryStorage(options);
+    },
+  };
+
+  return (
+    <PipelineStorageProvider value={new CompassPipelineStorage()}>
+      <FavoriteQueryStorageProvider value={favoriteQueryStorageProviderValue}>
+        <RecentQueryStorageProvider value={recentQueryStorageProviderValue}>
```
Comment (gribnoysup) [inline packages/compass/src/app/index.tsx:142]: Same as above, now that this code is part of component render, let's ensure we don't re-create instances

```suggestion
  const favoriteQueryStorageProviderValue: FavoriteQueryStorageAccess = useRef({
    getStorage(options) {
      return new CompassFavoriteQueryStorage(options);
    },
  });
  const recentQueryStorageProviderValue: RecentQueryStorageAccess = useRef({
    getStorage(options) {
      return new CompassRecentQueryStorage(options);
    },
  });
  const pipelineStorage = useRef(new CompassPipelineStorage());
  
  return (
    <PipelineStorageProvider value={pipelineStorage.current}>
      <FavoriteQueryStorageProvider value={favoriteQueryStorageProviderValue.current}>
        <RecentQueryStorageProvider value={recentQueryStorageProviderValue.current}>
```

**Review by gribnoysup — APPROVED** [review]
Looks good, just one small thing to address before merging

File: packages/compass/src/app/index.tsx (hunk 1 above)
Comment (mabaasit) [reply packages/compass/src/app/index.tsx]: added in [cde31be](https://github.com/mongodb-js/compass/pull/5512/commits/cde31be18d2876ae69685be93d307a825b171053)

**Pull request merged**